sycl/source/handler.cpp

```diff
@@ -368,11 +368,28 @@ event handler::finalize() {
         std::move(MArgs), std::move(CGData), MCGType, MCodeLoc));
     break;
   case detail::CG::Barrier:
-  case detail::CG::BarrierWaitlist:
-    CommandGroup.reset(new detail::CGBarrier(std::move(MEventsWaitWithBarrier),
-                                             std::move(CGData), MCGType,
-                                             MCodeLoc));
+  case detail::CG::BarrierWaitlist: {
+    if (auto GraphImpl = getCommandGraph(); GraphImpl != nullptr) {
+      // if no event to wait for was specified, we add all the previous
+      // nodes/events of the graph
```

Comment:
```suggestion
      // if no event to wait for was specified, we add all exit
      // nodes/events of the graph
```
Small clarification since we're not adding all nodes here but only the exit/leaf nodes.